File: 8/Java/inherit_overload.java
import java.io.*;
import java.lang.*;

class a //inheritance supports overloading in java but not in c++
{
	void g()
	{
		System.out.println("anil");
	}
}

class b extends a
{
	void g(int w)
	{
		System.out.println(10*w);
	}
}

class kapil
{
	public static void main(String[] args) {
		b y;
		y = new b();
		y.g(2);
	}
}

File: 8/C++/inherit_overload.cpp
#include <iostream>
using namespace std;

class a
{
	public: void g()
	{
		cout<<"anil"<<endl;
	}
};

class b: public a
{
	public: void g(int w)
	{
		cout<<10*w<<endl;
	}
};

int main()
{
	b y;
	y.g(2);
}